Source/Modules/app_view_models/Edit/Plugins/Sampler/SynthSamplerState.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+namespace app_view_models {
+namespace IDs {
+
+const juce::Identifier synthSamplerStateID("SYNTH_SAMPLER_STATE");
+const juce::Identifier synthSamplerCurFileID("synthSamplerCurFile");
+
+} // namespace IDs
+
+class SynthSamplerState : public juce::ValueTree::Listener,
+                          public FlaggedAsyncUpdater {
+  public:
+    SynthSamplerState(juce::ValueTree parent, int size);
+
+    juce::File getFile();
+    void setFile(juce::File newFile);
+
+    class SynthStateListener {
+      public:
+        virtual ~SynthStateListener() = default;
+        virtual void fileChanged(juce::File newFile) {}
+    };
+
+    void addListener(SynthStateListener *l);
+    void removeListener(SynthStateListener *l);
+
+    int listSize;
+
+  private:
+    // this stores the state of the list (the selected index)
+    juce::ValueTree synthSamplerValue;
+    tracktion::ConstrainedCachedValue<juce::String> curFilePath;
```

Comment:
this does not need to be a constrained cached value and can be a regular cached value

Reply:
chenged to juce::CachedValue